include/envoy/server/filter_config.h

```diff
@@ -58,6 +59,11 @@ class FactoryContext {
   virtual bool healthCheckFailed() PURE;
 
   /**
+   * @return HttpRouteManager& singleton for use by the entire server.
+   */
+  virtual HttpRouteManager& httpRouteManager() PURE;
```

Comment:
At the factory context level, we should not expose the entire manager. Only the ability to get a route provider by name. See below for interface changes.